src/clj/rems/application/model.clj

```diff
@@ -572,6 +572,20 @@
       (assoc :application/invited-members (set (vals (:application/invitation-tokens application))))
       (update :application/events (partial mapv #(dissoc % :invitation/token)))))
 
+(defn hide-answers-with-privacy [application roles]
+  (let [fields (get-in application [:application/form :form/fields])
+        fields-with-privacy (mapv (fn [field]
+                                    (if (or (some #{:applicant :member :decider :past-decider :handler :reporter}
+                                                  roles)
+                                            (= :public (:field/privacy field :public)))
+                                      (assoc field :field/private false)
```

Comment:
I wonder if it makes sense to call this enrichment or hiding ...

Reply:
hide sounds nice, but it could also be something generic like apply-privacy...